### ISSUE TITLE
Cannot use discrete AMD dGPU

### ISSUE THREAD

**Issue by yioannides** [minetest/minetest#15556]
### Luanti version

```true
Luanti v5.10.0 (flatpak) / no verbose on flatpak
```

### Irrlicht device

X11

### Operating system and version

Fedora Workstation 41

### CPU model

AMD Ryzen 7 7700X

### GPU model

AMD Radeon RX 6600

### Active renderer

OpenGL 4.6

### Summary

I cannot seem to be able to launch Luanti via discrete graphics card. The flatpak version only provides the "Launch with Integrated Graphics" when right-clicking on the icon, while the .rpm version does provide the "Launch with Dedicated Graphics Card" option, but launches with the integrated one, as seen on the performance screenshot below (the result is the same for both):

![Screenshot From 2024-12-14 21-06-56](https://github.com/user-attachments/assets/efaff95e-260b-4593-91e2-a86cdc7d34b9)
![Screenshot From 2024-12-14 21-08-57](https://github.com/user-attachments/assets/7eeb97f0-7c5b-4ad0-9c9d-1238eecd102e)

I don't know whether it's a Wayland issue or not, but Fedora 41 runs exclusively on Wayland. I would be more than happy to provide any additional information necessary, thank you!

### Steps to reproduce

1. Download via either flatpak or .rpm
2. Launch
3. Monitor the system resources

**Comment by SmallJoker** [comment]
I believe this is an issue caused by the Flatpak container. However:

> Download via either flatpak or .rpm

Does that mean that the issue appears for system-wide installs as well?

**Comment by yioannides** [comment]
That's correct, it also happens when installed via Fedora's native package repository. While this one has the "Launch with Discrete Graphics Card" options, the discrete card remains idle while the integrated graphics card constantly peaks at 100%.

**Comment by SmallJoker** [comment]
1. Start Luanti with the environment variable `DRI_PRIME=1`
   * More information: https://wiki.archlinux.org/title/PRIME#For_open_source_drivers_-_PRIME
   * Even more: https://wiki.gentoo.org/wiki/AMDGPU#Identifying_which_graphics_card_is_in_use
2. If you are on Wayland (i.e. using xwayland), retry with X11 directly (if installed) by switching in the login screen
3. If this issue is about debugging your setup like https://github.com/minetest/minetest/issues/14415#issuecomment-1963945964, then the forums might be a better place: https://forum.luanti.org/

**Comment by yioannides** [comment]
Okay, I launched Luanti via `DRI_PRIME=1 flatpak run net.minetest.Minetest`, started a new game and got some performance out of the dedicated graphics card, but for some reason, my integrated graphics card was also working hard:

![Screenshot From 2024-12-15 09-35-25](https://github.com/user-attachments/assets/d53fc5a8-0c38-4a37-9de3-48526bfde401)

EDIT: I guess launching it normally also makes the dedicated graphics card work some, but I don't understand what's going on with the integrated graphics card. This is the performance while simply left-clicking on the Luanti icon:

![Screenshot From 2024-12-15 09-38-51](https://github.com/user-attachments/assets/df63c24a-4b32-4aed-83f1-39a502dd5b7d)

How can I know for sure what card is running while on Luanti and how is it performing real-time? Also, the output of `DRI_PRIME=1 glxinfo | grep "OpenGL renderer"` is:

`OpenGL renderer string: AMD Radeon RX 6600 (radeonsi, navi23, LLVM 19.1.0, DRM 3.59, 6.11.11-300.fc41.x86_64)`

To be clear, I don't have any complains over performance, I was mostly curious seeing my system resources performing while on Luanti and the fact that there wasn't a graphics card option available in settings made me think that something could be wrong with my setup!
